src/state_point.F90

```diff
@@ -40,19 +41,20 @@ module state_point
 
   subroutine write_state_point()
 
-    integer                       :: i, j, k
-    integer                       :: n_order      ! loop index for moment orders
-    integer                       :: nm_order     ! loop index for Ynm moment orders
-    integer, allocatable          :: id_array(:)
-    integer, allocatable          :: key_array(:)
+    integer :: i, j, k
+    integer :: i_list, i_xs
+    integer :: n_order      ! loop index for moment orders
+    integer :: nm_order     ! loop index for Ynm moment orders
+    integer, allocatable :: id_array(:)
+    integer, allocatable :: key_array(:)
     integer(HID_T) :: file_id
     integer(HID_T) :: cmfd_group
     integer(HID_T) :: tallies_group, tally_group
     integer(HID_T) :: meshes_group, mesh_group
-    integer(HID_T) :: filter_group, moments_group
-    character(8)                  :: moment_name  ! name of moment (e.g, P3)
-    character(MAX_FILE_LEN)       :: filename
-    type(StructuredMesh), pointer :: meshp
+    integer(HID_T) :: filter_group
+    character(20), allocatable :: str_array(:)
+    character(MAX_FILE_LEN)    :: filename
+    type(RegularMesh), pointer :: meshp
```

Comment:
The PGI compiler has some weird bug where if you have a local variable that is the same name as a module, it sometimes produces an error... yes, only sometimes, and this was one of those times.